### PR TITLE
JBPM-9659: Upgrade deployments to WildFly 23

### PR DESCRIPTION
**Pull request by jesuino**
**JIRA**: 

[JBPM-9659](https://issues.redhat.com/browse/JBPM-9659)

**Related PRs*: 

https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/1693
https://github.com/kiegroup/kie-soup/pull/216
https://github.com/kiegroup/droolsjbpm-knowledge/pull/543
https://github.com/kiegroup/drools/pull/3757
https://github.com/kiegroup/optaplanner/pull/1463
https://github.com/kiegroup/lienzo-tests/pull/124
https://github.com/kiegroup/appformer/pull/1163
https://github.com/kiegroup/kie-uberfire-extensions/pull/122
https://github.com/kiegroup/jbpm/pull/1974
https://github.com/kiegroup/kie-jpmml-integration/pull/62
https://github.com/kiegroup/droolsjbpm-integration/pull/2539
https://github.com/kiegroup/kie-wb-playground/pull/115
https://github.com/kiegroup/kie-wb-common/pull/3660
https://github.com/kiegroup/drools-wb/pull/1506
https://github.com/kiegroup/jbpm-designer/pull/913
https://github.com/kiegroup/jbpm-work-items/pull/212
https://github.com/kiegroup/jbpm-wb/pull/1506
https://github.com/kiegroup/optaplanner-wb/pull/399
https://github.com/kiegroup/kie-wb-distributions/pull/1115
https://github.com/kiegroup/openshift-drools-hacep/pull/116
https://github.com/kiegroup/process-migration-service/pull/5
https://github.com/kiegroup/optaweb-employee-rostering/pull/662
https://github.com/kiegroup/optaweb-vehicle-routing/pull/535

### CODE REVIEW

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

**Comment by mbiarnes** [comment]
Jenkins retest this

File: uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/pom.xml [hunk 1]
```diff
@@ -177,6 +177,16 @@
     <dependency>
       <groupId>org.infinispan</groupId>
       <artifactId>infinispan-client-hotrod</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.sshd</groupId>
+          <artifactId>sshd-common</artifactId>
+        </exclusion>
```
Comment (tiagobento) [inline uberfire-extensions/uberfire-metadata/uberfire-metadata-commons-io/pom.xml:184]: @jesuino I'm not sure about this one. Can you please double-check?

File: uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/pom.xml [hunk 2]
```diff
@@ -96,6 +96,12 @@
     <dependency>
       <groupId>org.wildfly.security</groupId>
       <artifactId>wildfly-elytron</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.sshd</groupId>
+          <artifactId>sshd-common</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (tiagobento) [inline uberfire-extensions/uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/pom.xml:104]: @jesuino This one too.

File: dashbuilder/dashbuilder-distros/src/main/assembly/assembly-tomcat-8.xml [hunk 3]
```diff
@@ -34,22 +34,21 @@
 
         <include>com.h2database:h2</include>
         <include>ch.qos.cal10n:cal10n-api</include>
-        <include>javax.activation:activation</include>
-        <include>javax.enterprise:cdi-api</include>
-        <include>javax.inject:javax.inject</include>
+        <include>jakarta.activation:jakarta.activation-api</include>
+        <include>jakarta.enterprise:jakarta.enterprise.cdi-api</include>
+        <include>jakarta.inject:jakarta.inject-api</include>
         <include>javax.mail:mail</include>
```
Comment (tiagobento) [inline dashbuilder/dashbuilder-distros/src/main/assembly/assembly-tomcat-8.xml:40]: @jesuino Should this be `com.sun.mail:jakarta.mail`?

**Comment by mareknovotny** [comment]
jenkins retest this please

**Review by tiagobento — APPROVED** [review]

**Review by adrielparedes — APPROVED** [review]

**Review by tomasdavidorg — APPROVED** [review]

**Pull request merged**